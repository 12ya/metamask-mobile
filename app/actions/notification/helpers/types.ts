import { Notification } from '../../../util/notifications';
import notificationsAction from './constants';

interface PerformSignInRequest {
  type: typeof notificationsAction.PERFORM_SIGN_IN_REQUEST;
}

interface PerformSignInSuccess {
  type: typeof notificationsAction.PERFORM_SIGN_IN_SUCCESS;
  payload: { accessToken: string };
}

interface PerformSignInFailure {
  type: typeof notificationsAction.PERFORM_SIGN_IN_FAILURE;
  payload: { error: string };
}

interface PerformSignOutRequest {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_REQUEST;
}

interface PerformSignOutSuccess {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_SUCCESS;
}

interface PerformSignOutFailure {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_FAILURE;
  payload: { error: string };
}

interface EnableProfileSyncingRequest {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_REQUEST;
}

interface EnableProfileSyncingSuccess {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_SUCCESS;
}

interface EnableProfileSyncingFailure {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_FAILURE;
  payload: { error: string };
}

interface DisableProfileSyncingRequest {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_REQUEST;
}

interface DisableProfileSyncingSuccess {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_SUCCESS;
}

interface DisableProfileSyncingFailure {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_FAILURE;
  payload: { error: string };
}

interface EnableNotificationsServicesRequest {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_REQUEST;
}

interface EnableNotificationsServicesSuccess {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_SUCCESS;
}

interface EnableNotificationsServicesFailure {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_FAILURE;
  payload: { error: string };
}

interface DisableNotificationsServicesRequest {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_REQUEST;
}

interface DisableNotificationsServicesSuccess {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_SUCCESS;
}

interface DisableNotificationsServicesFailure {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_FAILURE;
  payload: { error: string };
}

interface SetFeatureAnnouncementsEnabledRequest {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST;
}

interface SetFeatureAnnouncementsEnabledSuccess {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS;
}

interface SetFeatureAnnouncementsEnabledFailure {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE;
  payload: { error: string };
}

interface SetSnapNotificationsEnabledRequest {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST;
}

interface SetSnapNotificationsEnabledSuccess {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS;
}

interface SetSnapNotificationsEnabledFailure {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE;
  payload: { error: string };
}

interface SetParticipateInMetaMetricsRequest {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_REQUEST;
}

interface SetParticipateInMetaMetricsSuccess {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_SUCCESS;
}

interface SetParticipateInMetaMetricsFailure {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_FAILURE;
  payload: { error: string };
}

interface SetMetamaskNotificationsFeatureSeenRequest {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST;
}

interface SetMetamaskNotificationsFeatureSeenSuccess {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS;
}

interface SetMetamaskNotificationsFeatureSeenFailure {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE;
  payload: { error: string };
}

interface FetchAndUpdateMetamaskNotificationsRequest {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST;
}

interface FetchAndUpdateMetamaskNotificationsSuccess {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS;
  payload: { notifications: Notification[] };
}

interface FetchAndUpdateMetamaskNotificationsFailure {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface MarkMetamaskNotificationsAsReadRequest {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST;
  payload: { notifications: Notification[] };
}

interface MarkMetamaskNotificationsAsReadSuccess {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS;
}

interface MarkMetamaskNotificationsAsReadFailure {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE;
  payload: { error: string };
}

interface CheckAccountsPresenceRequest {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_REQUEST;
  payload: { accounts: string[] };
}

interface CheckAccountsPresenceSuccess {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_SUCCESS;
  payload: { presence: Record<string, boolean> };
}

interface CheckAccountsPresenceFailure {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_FAILURE;
  payload: { error: string };
}

interface UpdateOnChainTriggersByAccountRequest {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST;
  payload: { accounts: string[] };
}

interface UpdateOnChainTriggersByAccountSuccess {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS;
}

interface UpdateOnChainTriggersByAccountFailure {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE;
  payload: { error: string };
}

interface DeleteOnChainTriggersByAccountRequest {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST;
  payload: { accounts: string[] };
}

interface DeleteOnChainTriggersByAccountSuccess {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS;
}

interface DeleteOnChainTriggersByAccountFailure {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE;
  payload: { error: string };
}

interface EnablePushNotificationsRequest {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface EnablePushNotificationsSuccess {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_SUCCESS;
}

interface EnablePushNotificationsFailure {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface DisablePushNotificationsRequest {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface DisablePushNotificationsSuccess {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_SUCCESS;
}

interface DisablePushNotificationsFailure {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface UpdateTriggerPushNotificationsRequest {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface UpdateTriggerPushNotificationsSuccess {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS;
  payload: { fcmToken: string };
}

interface UpdateTriggerPushNotificationsFailure {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

export type NotificationsActionTypes =
  | PerformSignInRequest
  | PerformSignInSuccess
  | PerformSignInFailure
  | PerformSignOutRequest
  | PerformSignOutSuccess
  | PerformSignOutFailure
  | EnableProfileSyncingRequest
  | EnableProfileSyncingSuccess
  | EnableProfileSyncingFailure
  | DisableProfileSyncingRequest
  | DisableProfileSyncingSuccess
  | DisableProfileSyncingFailure
  | EnablePushNotificationsRequest
  | EnablePushNotificationsSuccess
  | EnablePushNotificationsFailure
  | DisablePushNotificationsRequest
  | DisablePushNotificationsSuccess
  | DisablePushNotificationsFailure
  | SetFeatureAnnouncementsEnabledRequest
  | SetFeatureAnnouncementsEnabledSuccess
  | SetFeatureAnnouncementsEnabledFailure
  | SetSnapNotificationsEnabledRequest
  | SetSnapNotificationsEnabledSuccess
  | SetSnapNotificationsEnabledFailure
  | SetParticipateInMetaMetricsRequest
  | SetParticipateInMetaMetricsSuccess
  | SetParticipateInMetaMetricsFailure
  | SetMetamaskNotificationsFeatureSeenRequest
  | SetMetamaskNotificationsFeatureSeenSuccess
  | SetMetamaskNotificationsFeatureSeenFailure
  | FetchAndUpdateMetamaskNotificationsRequest
  | FetchAndUpdateMetamaskNotificationsSuccess
  | FetchAndUpdateMetamaskNotificationsFailure
  | MarkMetamaskNotificationsAsReadRequest
  | MarkMetamaskNotificationsAsReadSuccess
  | MarkMetamaskNotificationsAsReadFailure
  | CheckAccountsPresenceRequest
  | CheckAccountsPresenceSuccess
  | CheckAccountsPresenceFailure
  | UpdateOnChainTriggersByAccountRequest
  | UpdateOnChainTriggersByAccountSuccess
  | UpdateOnChainTriggersByAccountFailure
  | DeleteOnChainTriggersByAccountRequest
  | DeleteOnChainTriggersByAccountSuccess
  | DeleteOnChainTriggersByAccountFailure
  | UpdateTriggerPushNotificationsRequest
  | UpdateTriggerPushNotificationsSuccess
  | UpdateTriggerPushNotificationsFailure
  | EnableNotificationsServicesRequest
  | EnableNotificationsServicesSuccess
  | EnableNotificationsServicesFailure
  | DisableNotificationsServicesRequest
  | DisableNotificationsServicesSuccess
  | DisableNotificationsServicesFailure;
