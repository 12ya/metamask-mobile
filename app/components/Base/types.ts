export interface TextProps {
  /**
   * Resets default style
   */
  reset?: boolean;
  /**
   * Align text to center
   */
  centered?: boolean;
  /**
   * Align text to right
   */
  right?: boolean;
  /**
   * Makes text bold
   */
  bold?: boolean;
  /**
   * Makes text green
   */
  green?: boolean;
  /**
   * Makes text black
   */
  black?: boolean;
  /**
   * Makes text blue
   */
  blue?: boolean;
  /**
   * Makes text grey
   */
  grey?: boolean;
  /**
   * Makes text red
   */
  red?: boolean;
  /**
   * Makes text orange
   */
  orange?: boolean;
  /**
   * Makes text fontPrimary color
   */
  primary?: boolean;
  /**
   * Makes text muted color
   */
  muted?: boolean;
  /**
   * Makes text italic and tight
   * used in disclaimers
   */
  disclaimer?: boolean;
  /**
   * Makes text black and bigger
   * Used in modals
   */
  modal?: boolean;
  /**
   * Makes text with bigger line height
   * Used in modals with information text
   */
  infoModal?: boolean;
  /**
   * Makes text small
   */
  small?: boolean;
  /**
   * Makes text big
   */
  big?: boolean;
  /**
   * Makes text even bigger
   */
  bigger?: boolean;
  /**
   * Makes text uppercase
   */
  upper?: boolean;
  /**
   * Applies a link style
   */
  link?: boolean;
  /**
   * Applies a strikethrough decoration
   */
  strikethrough?: boolean;
  /**
   * Applies a underline decoration
   */
  underline?: boolean;
  /**
   * Removes the vertical margin
   */
  noMargin?: boolean;
  /**
   * Any other external style defined in props will be applied
   */
  style?: any;
}
