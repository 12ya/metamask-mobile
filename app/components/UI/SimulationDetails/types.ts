import { Hex } from '@metamask/utils';
import { BigNumber } from 'bignumber.js';

export enum AssetType {
  Native = 'NATIVE',
  ERC20 = 'ERC20',
  ERC721 = 'ERC721',
  ERC1155 = 'ERC1155',
}

export const NATIVE_ASSET_IDENTIFIER: NativeAssetIdentifier = {
  type: AssetType.Native,
};

/**
 * Describes an amount of fiat.
 */
export const FIAT_UNAVAILABLE = null;
export type FiatAmountAvailable = number;
export type FiatAmount = FiatAmountAvailable | typeof FIAT_UNAVAILABLE;

/**
 * Identifies the native asset of a chain.
 */
export interface NativeAssetIdentifier {
  type: AssetType.Native;
  address?: undefined;
  tokenId?: undefined;
}

/**
 * Uniquely identifies a token asset on a chain.
 */
export interface TokenAssetIdentifier {
  type: Exclude<AssetType, AssetType.Native>;
  address: Hex;
  tokenId?: Hex;
}

export type AssetIdentifier = Readonly<
  NativeAssetIdentifier | TokenAssetIdentifier
>;

/**
 * Describes a change in an asset's balance to a user's wallet.
 */
export type BalanceChange = Readonly<{
  /**
   * The asset identifier for the balance change.
   */
  asset: AssetIdentifier;

  /**
   * The quantity of asset tokens, expressed as a decimal value.
   *
   * This property represents the amount of tokens, taking into account the
   * number of decimals supported by the asset. The value can be positive
   * (increase) or negative (decrease).
   *
   * Example: If an asset supports 18 decimals, an `amount` of 1.5 represents
   * 1.5 tokens, or more precisely, 1.5 * 10^18 of the smallest divisible unit.
   */
  amount: BigNumber;
}>;

/** Token standards supported by simulation. */
export enum SimulationTokenStandard {
  erc20 = 'erc20',
  erc721 = 'erc721',
  erc1155 = 'erc1155',
}

/** Simulation data concerning an updated token. */
export interface SimulationToken {
  /** The token's contract address. */
  address: Hex;

  /** The standard of the token. */
  standard: SimulationTokenStandard;

  /** The ID of the token if supported by the standard. */
  id?: Hex;
}

/** Simulation data concerning an update to a native or token balance. */
export interface SimulationBalanceChange {
  /** The balance before the transaction. */
  previousBalance: Hex;

  /** The balance after the transaction. */
  newBalance: Hex;

  /** The difference in balance. */
  difference: Hex;

  /** Whether the balance is increasing or decreasing. */
  isDecrease: boolean;
}

/** Simulation data concerning a change to the a token balance. */
export type SimulationTokenBalanceChange = SimulationToken &
  SimulationBalanceChange;

export enum SimulationErrorCode {
  ChainNotSupported = 'chain-not-supported',
  Disabled = 'disabled',
  InvalidResponse = 'invalid-response',
  Reverted = 'reverted',
}

/** Error data for a failed simulation. */
export interface SimulationError {
  /** Error code to identify the error type. */
  code?: string | number;

  /** Error message to describe the error. */
  message?: string;
}

/** Simulation data for a transaction. */
export interface SimulationData {
  /** Error data if the simulation failed or the transaction reverted. */
  error?: SimulationError;

  /** Data concerning a change to the user's native balance. */
  nativeBalanceChange?: SimulationBalanceChange;

  /** Data concerning a change to the user's token balances. */
  tokenBalanceChanges: SimulationTokenBalanceChange[];
}
