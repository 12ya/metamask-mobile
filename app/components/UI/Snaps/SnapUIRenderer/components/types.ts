import { JSXElement, MaybeArray } from '@metamask/snaps-sdk/jsx';

export interface UIComponentParams<T extends JSXElement> {
  map: Record<string, number>;
  element: T;
  form?: string;
}

export interface UIComponent {
  element: string;
  props?: Record<string, unknown>;
  children?: MaybeArray<UIComponent | string>;
  key?: string;
}

export type UIComponentFactory<T extends JSXElement> = (
  params: UIComponentParams<T>,
) => UIComponent;
