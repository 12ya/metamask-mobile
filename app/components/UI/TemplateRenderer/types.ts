import { SafeComponentListValues } from './SafeComponentList';

export type Sections = string | SectionShape | (string | SectionShape)[];

export interface SectionShape {
  element: keyof SafeComponentListValues;
  key: string;
  props?: Record<string, unknown>;
  children?: Sections;
}
