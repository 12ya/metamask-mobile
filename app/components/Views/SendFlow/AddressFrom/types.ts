export interface SFAddressFromProps {
    accountAddress: string;
    accountName: string;
    accountBalance: string | undefined;
    updateAccountInfo: ({accountAddress, accountName, accountBalance, balanceIsZero}: {accountAddress: string, accountName: string, accountBalance: string, balanceIsZero: boolean} ) => void;
}

export interface SelectedAsset {
    address: string;
    isETH: boolean;
    logo: string;
    name: string;
    symbol: string;
}