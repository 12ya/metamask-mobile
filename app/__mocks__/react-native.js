import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
