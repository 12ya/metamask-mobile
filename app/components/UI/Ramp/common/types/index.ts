export type { Region } from '../../../../../redux/slices/fiatOrders/types';
export { RampType } from '../../../../../redux/slices/fiatOrders/types';

export enum PROVIDER_LINKS {
  HOMEPAGE = 'Homepage',
  PRIVACY_POLICY = 'Privacy Policy',
  SUPPORT = 'Support',
}

export * from './analytics';
