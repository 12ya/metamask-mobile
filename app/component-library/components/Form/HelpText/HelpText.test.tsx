// Third party dependencies.
import React from 'react';
import { render } from '@testing-library/react-native';

// Internal dependencies.
import HelpText from './HelpText';
import {
  DEFAULT_HELPTEXT_TEXT_VARIANT,
  HELPTEXT_TEST_ID,
  SAMPLE_HELPTEXT_TEXT,
  TEXT_COLOR_BY_HELPTEXT_SEVERITY,
} from './HelpText.constants';
import { HelpTextSeverity } from './HelpText.types';

describe('HelpText', () => {
  it('should render default settings correctly', () => {
    const { toJSON } = render(<HelpText>{SAMPLE_HELPTEXT_TEXT}</HelpText>);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render HelpText', () => {
    const { toJSON } = render(<HelpText>{SAMPLE_HELPTEXT_TEXT}</HelpText>);
    const helpTextComponent = wrapper.findWhere(
      (node) => node.prop('testID') === HELPTEXT_TEST_ID,
    );
    expect(helpTextComponent.exists()).toBe(true);
  });
  it('should render the given severity color', () => {
    const testSeverity = HelpTextSeverity.Error;
    const { toJSON } = render(
      <HelpText severity={testSeverity}>{SAMPLE_HELPTEXT_TEXT}</HelpText>,
    );
    const helpTextComponent = wrapper.findWhere(
      (node) => node.prop('testID') === HELPTEXT_TEST_ID,
    );
    expect(helpTextComponent.props().color).toBe(
      TEXT_COLOR_BY_HELPTEXT_SEVERITY[testSeverity],
    );
  });
  it('should render the given text with the appropriate variant', () => {
    const { toJSON } = render(<HelpText>{SAMPLE_HELPTEXT_TEXT}</HelpText>);
    const helpTextComponent = wrapper.findWhere(
      (node) => node.prop('testID') === HELPTEXT_TEST_ID,
    );
    expect(helpTextComponent.props().children).toBe(SAMPLE_HELPTEXT_TEXT);
    expect(helpTextComponent.props().variant).toBe(
      DEFAULT_HELPTEXT_TEXT_VARIANT,
    );
  });
});
