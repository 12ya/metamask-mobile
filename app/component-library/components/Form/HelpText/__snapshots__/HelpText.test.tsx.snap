// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText should render default settings correctly 1`] = `
<Text
  accessibilityRole="text"
  style={
    {
      "color": "#141618",
      "fontFamily": "Euclid Circular B",
      "fontSize": 12,
      "fontWeight": "400",
      "letterSpacing": 0,
      "lineHeight": 20,
    }
  }
  testID="helptext"
>
  Sample HelpText text
</Text>
`;
